src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -89,18 +79,15 @@ private void moveNodeToLast(Entry<K, V> entry) {
     public void put(K key, V value) {
         if (data.containsKey(key)) {
             final Entry<K, V> existingEntry = data.get(key);
-            existingEntry.setValue(value);
             moveNodeToLast(existingEntry);
+            existingEntry.setValue(value); 
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rishabhbohra3/Algo-tutorial-test/pull/6#issuecomment-2213714826>#6f5971</a><span> - 07/08/2024, 04:44 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The line 'existingEntry.setValue(value);' should be followed by updating the position of the entry in the LRU cache to reflect its recent use. Failing to do so can lead to incorrect eviction of recently used entries.
</div>
<div id="fix">
<b>Fix:</b> Ensure that the entry is moved to the end of the cache to reflect its recent use after updating its value.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -83,0 +84,1 @@
 +            moveNodeToLast(existingEntry);
```
</div>
</details>
</div>
<br>


</div>